### PR TITLE
[MPA] Enable SSD and YOLOX model template

### PR DESCRIPTION
**Pull request by jaegukhyun**
- Add SSD and YOLOX model
- Change template_experimental.yaml to template.yaml
- Change optimization hyperparameters
- Change learning schedule hyperparameters

### CODE REVIEW

File: external/model-preparation-algorithm/configs/detection/cspdarknet_yolox_cls_incr/model.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+_base_ = [
+  '../../../submodule/recipes/stages/_base_/models/detectors/yolox.custom.py'
+]
```
Comment (harimkang) [inline external/model-preparation-algorithm/configs/detection/cspdarknet_yolox_cls_incr/model.py]: All the files in the PR also seem to need flake8 check.

File: external/model-preparation-algorithm/configs/detection/cspdarknet_yolox_cls_incr/template.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Description.
```
Comment (harimkang) [inline external/model-preparation-algorithm/configs/detection/cspdarknet_yolox_cls_incr/template.yaml]: First, erase "template.yaml" and leave "template_experimental.yam'". The name change will be carried out in another PR.

Comment (harimkang) [reply external/model-preparation-algorithm/configs/detection/cspdarknet_yolox_cls_incr/template.yaml]: Please apply this to all model templates the same.

File: external/model-preparation-algorithm/configs/detection/mobilenetv2_ssd_cls_incr/model.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+_base_ = [
+  '../../../submodule/samples/cfgs/models/backbones/ote_mobilenet_v2_w1.yaml',
+  '../../../submodule/recipes/stages/_base_/models/detectors/ssd.custom.py'
+]
+
+model = dict(
+    backbone=dict(out_indices=(4, 5,))
```
Comment (harimkang) [inline external/model-preparation-algorithm/configs/detection/mobilenetv2_ssd_cls_incr/model.py:7]: What does this mean?

File: external/model-preparation-algorithm/mpa_tasks/apis/detection/task.py [hunk 4]
```diff
@@ -177,7 +177,15 @@ def _init_recipe(self):
             # recipe = os.path.join(recipe_root, 'pretrain.yaml')
             raise NotImplementedError(f'train type {train_type} is not implemented yet.')
         elif train_type == TrainType.Incremental:
-            recipe = os.path.join(recipe_root, 'imbalance.py')
+            if self._model_name.split('-')[0] == 'YOLOX':
```
Comment (harimkang) [inline external/model-preparation-algorithm/mpa_tasks/apis/detection/task.py]: I also left a comment related to the PR of the MPA side. 
Instead of separating the recipe according to the model, I think you can process ignore=false and put the data_pipeline directly into the OTE model config folder if necessary. 

For an example, refer to instance-segmentation.

**Review by harimkang — CHANGES_REQUESTED** [review]

File: external/model-preparation-algorithm/configs/detection/mobilenetv2_ssd_cls_incr/model.py (hunk 3 above)
Comment (jaegukhyun) [reply external/model-preparation-algorithm/configs/detection/mobilenetv2_ssd_cls_incr/model.py:7]: I just followed ATSS's model.py, mmdetection/configs/custom-object-detection/gen3_mobilenetV2_SSD/model.py.
I guess this code may specify which backbone's features would be used by neck or bbox_head.

File: external/model-preparation-algorithm/mpa_tasks/apis/detection/task.py [hunk 5]
```diff
@@ -177,7 +177,11 @@ def _init_recipe(self):
             # recipe = os.path.join(recipe_root, 'pretrain.yaml')
             raise NotImplementedError(f'train type {train_type} is not implemented yet.')
         elif train_type == TrainType.Incremental:
-            recipe = os.path.join(recipe_root, 'imbalance.py')
+            if self._model_name.split('-')[0] in ['ATSS', 'SSD', 'YOLOX']:
```
Comment (harimkang) [inline external/model-preparation-algorithm/mpa_tasks/apis/detection/task.py]: Is there a reason why this is necessary? Also, VFNet seems to be missing here.

**Review by harimkang — CHANGES_REQUESTED** [review]

**Comment by harimkang** [comment]
MPA api & pre-commit tests for this PR: 
https://ci-ote.iotg.sclab.intel.com/job/ote/job/pr-model-preparation-algorithm-test/39/

**Review by harimkang — APPROVED** [review]
MPA validation report ALL PASSED (API & precommit only)
http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=pr-model-preparation-algorithm-test-39&environment=iotg-dev-workstation-12

**Review by goodsong81 — APPROVED** [review]
LGTM.
